Compiler set in CC variable not found, when used with rules from rules_foreign_cc

Hi

I'm trying to use the llvm_toolchain together with `rules_foreign_cc` and get the following error building a target using `cmake_external`:

```
Bazel external C/C++ Rules #0.0.8. Building library 'jansson'
Environment:______________
TMPDIR=/var/folders/y5/lzx7l8210lnd52jz23_tbztm0000gn/T/
EXT_BUILD_ROOT=/private/var/tmp/_bazel_subotic/5261f9905fd7fd27aa8a4af6c3ff2ca2/sandbox/darwin-sandbox/3/execroot/swiss_dasch_skunkworks
INSTALLDIR=/private/var/tmp/_bazel_subotic/5261f9905fd7fd27aa8a4af6c3ff2ca2/sandbox/darwin-sandbox/3/execroot/swiss_dasch_skunkworks/bazel-out/darwin-dbg/bin/third_party/jansson
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
PATH=/private/var/tmp/_bazel_subotic/5261f9905fd7fd27aa8a4af6c3ff2ca2/sandbox/darwin-sandbox/3/execroot/swiss_dasch_skunkworks:/usr/gnu/bin:/usr/local/bin:/bin:/usr/bin:.
BUILD_TMPDIR=/var/folders/y5/lzx7l8210lnd52jz23_tbztm0000gn/T/tmp.SOcWGktR
PWD=/private/var/tmp/_bazel_subotic/5261f9905fd7fd27aa8a4af6c3ff2ca2/sandbox/darwin-sandbox/3/execroot/swiss_dasch_skunkworks
EXT_BUILD_DEPS=/var/folders/y5/lzx7l8210lnd52jz23_tbztm0000gn/T/tmp.8oafiMc7
SHLVL=2
BUILD_LOG=bazel-out/darwin-dbg/bin/third_party/jansson/logs/CMake.log
BUILD_SCRIPT=bazel-out/darwin-dbg/bin/third_party/jansson/logs/CMake_script.sh
_=/usr/bin/env
__________________________
sh: /bin/ps: Operation not permitted
CMake Error at /usr/local/Cellar/cmake/3.19.1/share/cmake/Modules/CMakeDetermineCCompiler.cmake:49 (message):
  Could not find compiler set in environment variable CC:

  
  /private/var/tmp/_bazel_subotic/5261f9905fd7fd27aa8a4af6c3ff2ca2/sandbox/darwin-sandbox/3/execroot/swiss_dasch_skunkworks/external/llvm_toolchain/bin/cc_wrapper.sh.
Call Stack (most recent call first):
  CMakeLists.txt:50 (project)


CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/var/folders/y5/lzx7l8210lnd52jz23_tbztm0000gn/T/tmp.SOcWGktR/CMakeFiles/CMakeOutput.log".
``` 

I see, that the CC variable is (correctly?) populated with `.../external/llvm_toolchain/bin/cc_wrapper.sh`, but that CMake does not like it? Is there something I'm missing? Any hints are welcome.

Operating system: macOS Big Sur
LLVM version: 11.0.0
bazel-toolchain: tag 0.5.7

--- a/.bazelrc
+++ b/.bazelrc
@@ -1,0 +1,1 @@
+build --incompatible_enable_cc_toolchain_resolution

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -101,3 +101,36 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 go_rules_dependencies()
 
 go_register_toolchains(version = "1.17")
+
+# For testing rules_foreign_cc.
+# See https://bazelbuild.github.io/rules_foreign_cc/0.6.0/cmake.html
+
+http_archive(
+    name = "rules_foreign_cc",
+    sha256 = "69023642d5781c68911beda769f91fcbc8ca48711db935a75da7f6536b65047f",
+    strip_prefix = "rules_foreign_cc-0.6.0",
+    url = "https://github.com/bazelbuild/rules_foreign_cc/archive/0.6.0.tar.gz",
+)
+
+load("@rules_foreign_cc//foreign_cc:repositories.bzl", "rules_foreign_cc_dependencies")
+
+rules_foreign_cc_dependencies()
+
+_ALL_CONTENT = """\
+filegroup(
+    name = "all_srcs",
+    srcs = glob(["**"]),
+    visibility = ["//visibility:public"],
+)
+"""
+
+http_archive(
+    name = "pcre",
+    build_file_content = _ALL_CONTENT,
+    sha256 = "0b8e7465dc5e98c757cc3650a20a7843ee4c3edf50aaf60bb33fd879690d2c73",
+    strip_prefix = "pcre-8.43",
+    urls = [
+        "https://mirror.bazel.build/ftp.pcre.org/pub/pcre/pcre-8.43.tar.gz",
+        "https://ftp.pcre.org/pub/pcre/pcre-8.43.tar.gz",
+    ],
+)

--- a/tests/foreign/BUILD.bazel
+++ b/tests/foreign/BUILD.bazel
@@ -1,0 +1,28 @@
+# Copyright 2021 The Bazel Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake")
+
+# See https://bazelbuild.github.io/rules_foreign_cc/0.6.0/cmake.html
+cmake(
+    name = "pcre",
+    cache_entries = {
+        "CMAKE_C_FLAGS": "-fPIC",
+    },
+    lib_source = "@pcre//:all_srcs",
+    out_static_libs = ["libpcre.a"],
+)
+
+# A smaller test would be something like
+# https://github.com/bazelbuild/rules_foreign_cc/blob/0.6.0/examples/cmake_hello_world_lib/binary/BUILD.bazel

--- a/tests/scripts/run_external_tests.sh
+++ b/tests/scripts/run_external_tests.sh
@@ -54,6 +54,7 @@ if [[ "${TEST_MIGRATION:-}" ]]; then
   test_args+=("--incompatible_disallow_struct_provider_syntax=false")
 fi
 "${bazel}" --bazelrc=/dev/null test "${test_args[@]}" \
+  //tests/foreign:pcre \
   @openssl//:libssl \
   $("${bazel}" query 'attr(timeout, short, tests(@com_google_absl//absl/...))') \
-  $("${bazel}" query 'tests(@io_bazel_rules_go//tests/core/cgo:all) except @io_bazel_rules_go//tests/core/cgo:cc_libs_test @io_bazel_rules_go//tests/core/cgo:opts_test')
+  $("${bazel}" query 'tests(@io_bazel_rules_go//tests/core/cgo:all) except set(@io_bazel_rules_go//tests/core/cgo:cc_libs_test @io_bazel_rules_go//tests/core/cgo:opts_test)')

--- a/tests/scripts/run_tests.sh
+++ b/tests/scripts/run_tests.sh
@@ -61,4 +61,4 @@ if [[ "${TEST_MIGRATION:-}" ]]; then
   # This flag is not quite ready -- https://github.com/bazelbuild/bazel/issues/7347
   test_args+=("--incompatible_disallow_struct_provider_syntax=false")
 fi
-"${bazel}"  --bazelrc=/dev/null test "${test_args[@]}" //...
+"${bazel}"  --bazelrc=/dev/null test "${test_args[@]}" //tests:all

